ospd_openvas/notus.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2021 Greenbone Networks GmbH
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+from typing import Callable
+from threading import Timer
+
+from ospd_openvas.messages.result import ResultMessage
+
+logger = logging.getLogger(__name__)
+
+
+class Notus:
+    def __init__(self, report_func: Callable[[list, str], bool]) -> None:
+        self._results = {}
+        self._report_func = report_func
+
+    def report_results(self, scan_id: str) -> None:
+        results = self._results.pop(scan_id)
+        if not self._report_func(results, scan_id):
+            logger.warning(
+                "Unable to report %d notus results for scan id %s.",
+                len(results),
+                scan_id,
+            )
+
+    def result_handler(self, res_msg: ResultMessage) -> None:
+
+        result = res_msg.serialize()
+        scan_id = result.pop("scan_id")
+        timer = None
+
+        if not scan_id in self._results:
+            self._results[scan_id] = []
+            timer = Timer(0.25, self.report_results, [scan_id])
```

Comment:
how about using a parameter for the timer value in case we want to check out, what value is best? (or a config file value)?

Reply:
IMHO I don't think the timer value matters that much and I don't like configuring this time, as there can also happen bugs, if someone set this time too high. So introducing a config for something that is not user relevant, but can cause bugs when set manually sounds dangerous. If we wan to find out the best value for this, the developer can test it on their own by changing this value(?)